neo/io/nwbio.py

```diff
@@ -0,0 +1,823 @@
+        signal = self._timeseries.data[i_start: i_stop]
+        if self.sampling_rate is None:
+            return IrregularlySampledSignal(
+                        self._timeseries.timestamps[i_start:i_stop] * pq.s,
+                        signal,
+                        units=self.units,
+                        t_start=sig_t_start,
+                        sampling_rate=self.sampling_rate,
+                        name=self.name,
+                        description=self.description,
+                        array_annotations=None,
+                        **self.annotations)  # todo: timeseries.control / control_description
+
+        else:
+            return AnalogSignal(
+                        signal,
+                        units=self.units,
+                        t_start=sig_t_start,
+                        sampling_rate=self.sampling_rate,
+                        name=self.name,
+                        description=self.description,
+                        array_annotations=None,
+                        **self.annotations)  # todo: timeseries.control / control_description
+
+
+class EventProxy(BaseEventProxy):
+
+    def __init__(self, timeseries, nwb_group):
+        self._timeseries = timeseries
+        self.name = timeseries.name
+        self.annotations = {"nwb_group": nwb_group}
+        self.description = try_json_field(timeseries.description)
+        if isinstance(self.description, dict):
+            self.annotations.update(self.description)
+            self.description = None
+        self.shape = self._timeseries.data.shape
+
+    def load(self, time_slice=None, strict_slicing=True):
+        """
+        *Args*:
+            :time_slice: None or tuple of the time slice expressed with quantities.
+                            None is the entire signal.
+            :strict_slicing: True by default.
+                Control if an error is raised or not when one of the time_slice members
+                (t_start or t_stop) is outside the real time range of the segment.
+        """
+        if time_slice:
+            raise NotImplementedError("todo")
+        else:
+            times = self._timeseries.timestamps[:]
+            labels = self._timeseries.data[:]
+        return Event(times * pq.s,
+                     labels=labels,
+                     name=self.name,
+                     description=self.description,
+                     **self.annotations)
+
+
+class EpochProxy(BaseEpochProxy):
+
+    def __init__(self, epochs_table, epoch_name=None, index=None):
+        self._epochs_table = epochs_table
+        if index is not None:
+            self._index = index
+            self.shape = (index.sum(),)
+        else:
+            self._index = slice(None)
+            self.shape = epochs_table.n_rows  # untested, just guessed that n_rows exists
+        self.name = epoch_name
+
+    def load(self, time_slice=None, strict_slicing=True):
```

Comment:
time_slice appears to do nothing. How is this supposed to interact with `self._index`?

Reply:
this is a "todo"

Reply:
is this on the todo list? #1013

Reply:
this is part of #1013